### ISSUE TITLE
Tags accessibility update

### ISSUE THREAD

**Issue by cathystoli** [chanzuckerberg/sci-components#702]
**Issue:**
1. With SR, cannot interact with the filter tag to remove it. 

2. SR reads component as 'button' and does not mention that it is dismissable.

3. Cannot be utilized via keyboard navigation.

4. Does not have focus control indicator.

5. Pointer input target is very small and only clicking directly on the icon will remove the tag.

**Solution:**
1. Allow filter tag to be removed via SR.

2. What would be the appropriate term for the SR to read out to indicate that this is a tag and it is removable?

3. Allow component to be used via keyboard navigation.

4. Add focus indicators.

5. Increase pointer input target for the action to improve accessibility.
